### PR TITLE
support AppleClang with openmp

### PR DESCRIPTION
**Pull request by levyfan**
support #1492 
Tested with cmake 3.12.0-rc3 and Apple LLVM version 9.1.0 (clang-902.0.39.2) and libomp 5.0.1.

### Howto
* install libomp by homebrew `brew install libomp`
* download and install cmake >= 3.12.0
* mkdir build ; cd build
* cmake ..
* make -j

### CODE REVIEW

**Comment by henry0312** [comment]
Oh, I didn't know AppleClang has supported OpenMP. Good news!

**Comment by StrikerRUS** [comment]
master fails too :(

Travis problems... again!

**Comment by henry0312** [comment]
There is something wrong with OSX on Travis.
I don't know why `curl: (3) <url> malformed` happns.

```
==> Installing gcc
==> Downloading /bottles/gcc-8.1.0.sierra.bottle.1.tar.gz
curl: (3) <url> malformed
Error: Failed to download resource "gcc"
Download failed: /bottles/gcc-8.1.0.sierra.bottle.1.tar.gz
Warning: Bottle installation failed: building from source.
==> Downloading https://ftp.gnu.org/gnu/gcc/gcc-8.1.0/gcc-8.1.0.tar.xz
==> ../configure --build=x86_64-apple-darwin16.7.0 --prefix=/usr/local/Cellar/gc
==> make
```

**Comment by StrikerRUS** [comment]
Travis seems to be OK now. Rerunning builds.

File: CMakeLists.txt [hunk 1]
```diff
@@ -24,7 +24,7 @@ elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
     message(FATAL_ERROR "Insufficient Clang version")
   endif()
 elseif(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
-    message(FATAL_ERROR "AppleClang isn't supported. Please see https://github.com/Microsoft/LightGBM/blob/master/docs/Installation-Guide.rst#macos")
+    cmake_minimum_required(VERSION 3.12)
```
Comment (StrikerRUS) [inline CMakeLists.txt]: Maybe it's better to move this to the top of the file where other `cmake_minimum_required` are listed?

**Comment by chivee** [comment]
@levyfan , could you also help on the updating of Documents?

File: docs/Installation-Guide.rst [hunk 2]
```diff
@@ -87,7 +87,25 @@ Also you may want to read `gcc Tips <./gcc-Tips.rst>`__.
 macOS
 ~~~~~
 
-LightGBM depends on **OpenMP** for compiling, which isn't supported by Apple Clang.
+LightGBM can use Apple Clang, or GCC/G++ to build in macOS.
+
+Apple Clang
+***********
+
+1. Install `CMake`_ (3.12 or higher).
```
Comment (chivee) [inline docs/Installation-Guide.rst]: It will be nice if the certain version of AppleClang been specified.

Comment (levyfan) [reply docs/Installation-Guide.rst]: I am not very certain about the minimal version. I know 
macOS >= el_capitan is supported by https://github.com/Homebrew/homebrew-core/blob/master/Formula/libomp.rb

Comment (chivee) [reply docs/Installation-Guide.rst]: A reasonable guess would be 9.0 (since 10.12).

**Comment by chivee** [comment]
I think we could update the Travis for macOS to using AppleClang as default. I' ll create related PR

**Comment by StrikerRUS** [comment]
@chivee 
>I think we could update the Travis for macOS to using AppleClang as default. I' ll create related PR

Don't it too early, because CMake 3.12 is Release Candidate now? I mean, can we install it via brew at Travis?

**Comment by levyfan** [comment]
@StrikerRUS we could use brew --HEAD for cmake now and switch to stable release later.

**Comment by levyfan** [comment]
Why is `VSTS: LightGBM-Build-CI` still running? 😿

**Comment by guolinke** [comment]
refer to https://github.com/Microsoft/vsts-agent/issues/1699#issuecomment-404731721

**Pull request merged**